controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -559,6 +562,44 @@ func (r *BareMetalHostReconciler) actionMatchProfile(prov provisioner.Provisione
 	return actionComplete{}
 }
 
+func (r *BareMetalHostReconciler) actionPreparing(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	info.log.Info("preparing")
+
+	// Save provisioning settings.
+	provisioningSettings := info.host.Status.Provisioning.DeepCopy()
+	dirty, err := saveHostProvisioningSettings(info.host)
+	if err != nil {
+		return actionError{errors.Wrap(err, "Could not save the host provisioning settings")}
+	}
+
+	// Do prepare(manual clean).
+	provResult, started, err := prov.Prepare(dirty)
+	if err != nil {
+		return actionError{errors.Wrap(err, "error preparing host")}
+	}
+
+	if provResult.ErrorMessage != "" {
+		info.log.Info("handling cleaning error in controller")
+		clearHostProvisioningSettings(info.host)
+		return recordActionFailure(info, metal3v1alpha1.PreparationError, provResult.ErrorMessage)
+	}
+
+	if provResult.Dirty {
+		result := actionContinue{provResult.RequeueAfter}
+		if clearError(info.host) || (dirty && started) {
+			// If clearError return true, but started is false, restore provisioningSettings.
```

Comment:
This comment is not really clear to me, can you please elaborate it a little bit?

Reply:
This is just describing how the if statement below can come to be true.